src/go/k8s/controllers/core/redpandacluster_controller.go

```diff
@@ -0,0 +1,417 @@
+		cluster.Name + seedSuffix + "." +
+		cluster.Namespace +
+		".svc.cluster.local"
+	cfg.Redpanda.Directory = dataDirectory
+
+	cfgBytes, err := yaml.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+
+	return r.Create(ctx, &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:	cluster.Namespace,
+			Name:		cluster.Name + seedSuffix,
+			Labels:		cluster.Labels,
+		},
+		Data: map[string]string{
+			"redpanda.yaml": string(cfgBytes),
+		},
+	})
+}
+
+func copyConfig(
+	c *corev1alpha1.RedpandaConfig, cfgDefaults *config.RedpandaConfig,
+) config.RedpandaConfig {
+	rpcServerPort := c.RPCServer.Port
+	if c.RPCServer.Port == 0 {
+		rpcServerPort = cfgDefaults.RPCServer.Port
+	}
+
+	kafkaAPIPort := c.KafkaAPI.Port
+	if c.KafkaAPI.Port == 0 {
+		kafkaAPIPort = cfgDefaults.KafkaApi.Port
+	}
+
+	AdminAPIPort := c.AdminAPI.Port
+	if c.AdminAPI.Port == 0 {
+		AdminAPIPort = cfgDefaults.AdminApi.Port
+	}
+
+	return config.RedpandaConfig{
+		RPCServer: config.SocketAddress{
+			Address:	"0.0.0.0",
+			Port:		rpcServerPort,
+		},
+		AdvertisedRPCAPI: &config.SocketAddress{
+			Port: c.AdvertisedRPCAPI.Port,
+		},
+		KafkaApi: config.SocketAddress{
+			Address:	"0.0.0.0",
+			Port:		kafkaAPIPort,
+		},
+		AdvertisedKafkaApi: &config.SocketAddress{
+			Port: c.AdvertisedKafkaAPI.Port,
+		},
+		AdminApi: config.SocketAddress{
+			Address:	"0.0.0.0",
+			Port:		AdminAPIPort,
+		},
+		DeveloperMode:	c.DeveloperMode,
+	}
+}
+
+// nolint:funlen // The definition needs further refinement
+func (r *RedpandaClusterReconciler) createBootstrapStatefulSet(
+	ctx context.Context, cluster *corev1alpha1.RedpandaCluster,
+) error {
+	memory, exist := cluster.Spec.Resources.Limits["memory"]
+	if !exist {
+		memory = resource.MustParse("2Gi")
+	}
+
+	return r.Create(ctx, &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:	cluster.Namespace,
+			Name:		cluster.Name + seedSuffix,
+			Labels:		cluster.Labels,
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas:		pointer.Int32Ptr(1),
+			PodManagementPolicy:	appsv1.ParallelPodManagement,
+			Selector:		metav1.SetAsLabelSelector(cluster.Labels),
+			UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+				Type: appsv1.RollingUpdateStatefulSetStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:		cluster.Name + seedSuffix,
+					Namespace:	cluster.Namespace,
+					Labels:		cluster.Labels,
+				},
+				Spec: corev1.PodSpec{
+					SecurityContext: &corev1.PodSecurityContext{
+						FSGroup: pointer.Int64Ptr(fsGroup),
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name:	"datadir",
+							VolumeSource: corev1.VolumeSource{
+								PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+									ClaimName: "datadir",
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:	"redpanda",
+							Image:	"vectorized/redpanda:" + cluster.Spec.Version,
```

Comment:
In the future let's consider making the image name configurable?

Reply:
This is not a problem to change it now.